packages/accounts-controller/src/AccountsController.ts

```diff
@@ -348,10 +349,15 @@ export class AccountsController extends BaseController<
         metadata: {
           ...internalAccount.metadata,
           name:
-            existingAccount && existingAccount.metadata.name !== ''
-              ? existingAccount.metadata.name
-              : `${keyringTypeName} ${keyringAccountIndex + 1}`,
-          lastSelected: existingAccount?.metadata?.lastSelected,
+            this.#populateExistingMetadata(existingAccount?.id, 'name') ??
```

Comment:
Related to my other comment that `exitstingAccount?.id` could be `undefined`. I guess we can use `internalAccount.id` here directly, no? This way we know it's not "potentially" `undefined`?